jsonschema/_validators.py

```diff
@@ -29,8 +29,18 @@ def additionalProperties(validator, aP, instance, schema):
             for error in validator.descend(instance[extra], aP, path=extra):
                 yield error
     elif not aP and extras:
-        error = "Additional properties are not allowed (%s %s unexpected)"
-        yield ValidationError(error % _utils.extras_msg(extras))
+        if "patternProperties" in schema:
+            patterns = sorted(schema["patternProperties"].keys())
+            if len(extras) == 1:
+                verb = "does"
+            else:
+                verb = "do"
+            error = "%s %s not match any of the regexs: %s" % (
+                ", ".join(map(repr, sorted(extras))), verb, ", ".join(patterns))
```

Comment:
Any reason to `map` `repr` for the first argument but not for the patterns?

Still would be a bit confusing I think if patterns ended up having the empty string in it.